### PR TITLE
Change match syntax

### PR DESCRIPTION
**Pull request by odersky**
`match` is now treated like an alphabetic operator, it can be appear infix as
well as after `.`.

Motivation given in https://contributors.scala-lang.org/t/pre-sip-demote-match-keyword-to-a-method/2137/36?u=odersky

### CODE REVIEW

**Comment by odersky** [comment]
Best reviewed by looking at changed files directly, since there was a bit of going back of forth on the way.

**Comment by lihaoyi** [comment]
Since `match` now behaves for all intents and purposes like a method call, are we going to demote it from a keyword to an intrinsic method the same way `eq`, `ne`, `==`, etc. are? Otherwise it would be weird if `match` was the only keyword that you could use with or without a preceding `.`

**Comment by nafg** [comment]
Does this make match redundant with pipe?

**Comment by odersky** [comment]
Making `match` behave more like a method call is an option. But `match` will have to stay a reserved word in any case. It is used in a special way in match types and inline matches, neither of which can be expressed as simple method calls. So that weakens the case of making it a method, since it would not be a clean abstraction.

**Comment by martijnhoekstra** [comment]
> Making `match` behave more like a method call is an option. But `match` will have to stay a reserved word in any case. It is used in a special way in match types and inline matches, neither of which can be expressed as simple method calls. So that weakens the case of making it a method, since it would not be a clean abstraction.

which IMO is exactly the reason it shouldn't *look* like a method either.

File: docs/docs/reference/changed-features/match-syntax.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    } match {
+      case "empty" => 0
+      case "nonempty" => 1
+    }
+
+ 2. `match` may follow a period:
+
+     ```scala
+     if xsDefined
+        && xs.match {
```
Comment (hrhino) [inline docs/docs/reference/changed-features/match-syntax.md]: So that's why they call it "dotty".

**Pull request merged**